mozilla-mobile/android-components/components/feature/prompts/src/main/res/values/strings.xml

```diff
@@ -25,4 +25,30 @@
     <string name="mozac_feature_prompts_allow">Allow</string>
     <!-- Text of a negative button in dialog requesting to open a new window. -->
     <string name="mozac_feature_prompts_deny">Deny</string>
+    <!-- Title of the month chooser dialog. -->
+    <string name="mozac_feature_prompts_set_month">Pick a month</string>
+    <!-- January (3 character long), used on the month chooser dialog. -->
+    <string name="mozac_feature_prompts_jan">Jan</string>
```

Comment:
@Pike Isn't this going to be complicated for many locales (ie, shortening name months to 3 characters)?
Took a look around at our other products, and I don't know if there is a workaround in this case

Reply:
https://www.unicode.org/cldr/charts/35/by_type/date_&_time.gregorian.html#Months_-_abbreviated_-_Standalone has some bad counter examples, but yeah, this is something that we should confirm that it's going to work with "few characters, but maybe more than 3"

Reply:
@Amejia481 or @pocmo : Do you have any idea if we can have more than 3 characters for this? We'd need to update the localization comments if that's the case. Thanks!

Reply:
@Delphine having fewer won't be a problem, but having more than five could cause some issues with the actual layout. I will update the comments and the layout to adapt independently of the amount of charters.

This is where we are using the string.
![image](https://user-images.githubusercontent.com/773158/62230407-bb904400-b38f-11e9-96f2-25769a480d12.png)

Reply:
https://github.com/mozilla-mobile/android-components/issues/3975

Reply:
The same dialog should exist in Fennec - maybe we can compare how it was translated there?

Reply:
@pocmo  they are using a deprecated version :(

Reply:
[This is what they are using: ](https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/widget/DateTimePicker.java#685)  `DateUtils.getMonthString(int month, int abbrev) ` [it was deprecated in API level 17 ](https://developer.android.com/reference/android/text/format/DateUtils.html#getMonthString(int,%20int))

Reply:
There might be a corresponding method in ICU now

Reply:
With `ICU` do you mean [android.icu.text](https://developer.android.com/reference/android/icu/text/package-summary)?

Reply:
Yeah, though I don't know exact APIs. It does have all the data you're interested in, I guess the question is if they shell out just the abbreviated month.

Reply:
The deprecation notice mentions `SimpleDateFormat`- is that not an option?

Reply:
Our own translations will definitely be handy once we do locale switching and have translations for less common languages.

Reply:
It's worth testing, gecko for example ships icu datasets beyond the locales we ship

Reply:
What you thing about using? We could build the list of month with the output :

```kotlin
    val today = Calendar.getInstance()
    for( month in 0..11){
        today.set(Calendar.MONTH, month)
        val formatter = SimpleDateFormat("MMM", Locale.getDefault())
        println(formatter.format(today.time))
    }
```

**Output**
```
Jan
Feb
Mar
Apr
May
Jun
Jul
Aug
Sep
Oct
Nov
Dec
```

Reply:
**Locale.forLanguageTag("ES")**
```
ene
feb
mar
abr
may
jun
jul
ago
sep
oct
nov
dic
```

**Locale.GERMANY** 
```
Jan
Feb
Mär
Apr
Mai
Jun
Jul
Aug
Sep
Okt
Nov
Dez
```

**Locale.FRANCE**
```
janv.
févr.
mars
avr.
mai
juin
juil.
août
sept.
oct.
nov.
déc.
```

**Locale.KOREAN**
```
1월
2월
3월
4월
5월
6월
7월
8월
9월
10월
11월
12월
```

**Locale.CHINESE**
```
一月
二月
三月
四月
五月
六月
七月
八月
九月
十月
十一月
十二月
```

Reply:
Looks good. My favorite language for Focus was `fy_NL` - can it handle that? :D

Reply:
When I used `Locale.forLanguageTag("fy_NL")`, I doesn't find a locale for that, I'm not sure if  `Locale.forLanguageTag("NL")` is an equivalent? 

**Locale.forLanguageTag("NL")**
```
jan
feb
mrt
apr
mei
jun
jul
aug
sep
okt
nov
dec
```

Reply:
http://www.localeplanet.com/icu/fy-NL/index.html

Reply:
That's `nl` for Dutch. The expected result should be https://github.com/unicode-org/cldr/blob/f99e5f01cb00c321ea2239d99c56923ded53f6ea/common/main/fy.xml#L2084-L2097.

Do we know what https://developer.android.com/reference/android/icu/text/DateFormat.html#getAvailableLocales() returns?

Reply:
This table seems to be interesting too, our min. SDK version of 21 supports ICU 53 / CLDR 25:
https://developer.android.com/guide/topics/resources/internationalization